packages/react/src/components/DateTimePicker/DateTimePicker.jsx

```diff
@@ -787,13 +795,60 @@ const DateTimePicker = ({
     setAbsoluteValue(newAbsolute);
   };
 
+  const isValidDate = (d, time) => {
```

Comment:
We are getting there :-) This check will however incorrectly return true for any time value following `nn:nn`, e.g. the value `99:99`. If you instead of splitting the time before and and pass in the full time string you could use the same regex used in the TimePickerSpinner.jsx:

```
const isValid24HoursRegex = /^([01][0-9]|2[0-3]):([0-5][0-9])$/;
const isValidTime = isValid24HoursRegex.test(time);
```

Ideally this would not be duplicated but in a separate file and imported int both components..

Reply:
Utilizing the regrex in the isValidDate function not. I was not sure where to put the regrex and was not sure where to put a new file if I made one for me to import it for both components, so I copied it into the DateTimePicker component file.